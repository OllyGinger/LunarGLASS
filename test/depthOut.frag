#version 130

varying vec4 Color;
varying float Depth;

out vec4 foo;

void main()
{
    gl_FragDepth = Depth;
    gl_FragColor = Color;
    foo = Color;
}
