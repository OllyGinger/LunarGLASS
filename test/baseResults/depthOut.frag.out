
Top IR:
; ModuleID = 'Glslang'

@gl_FragDepth = global float 0.000000e+00
@Depth = global float 0.000000e+00
@gl_FragColor = global <4 x float> zeroinitializer
@Color = global <4 x float> zeroinitializer

define fastcc void @main() {
entry:
  %gl_FragDepth = load float* @Depth
  store float %gl_FragDepth, float* @gl_FragDepth
  %gl_FragColor = load <4 x float>* @Color
  store <4 x float> %gl_FragColor, <4 x float>* @gl_FragColor
  br label %stage-epilogue

stage-epilogue:                                   ; preds = %entry
  br label %stage-exit

stage-exit:                                       ; preds = %stage-epilogue
  ret void
}

!gla.entrypoint = !{!0}
!gla.outputs = !{!1, !3}
!gla.inputs = !{!5, !7}

!0 = metadata !{metadata !"main", i32 15}
!1 = metadata !{metadata !"gl_FragDepth", i32 11, float* @gl_FragDepth_typeProxy, metadata !2}
!2 = metadata !{i32 0, i32 0, i32 1024, null, i32 0}
!3 = metadata !{metadata !"gl_FragColor", i32 7, <4 x float>* @gl_FragColor_typeProxy, metadata !4}
!4 = metadata !{i32 0, i32 0, i32 1026, null, i32 0}
!5 = metadata !{metadata !"Depth", i32 1, float* @Depth_typeProxy, metadata !6}
!6 = metadata !{i32 0, i32 0, i32 1025, null, i32 1}
!7 = metadata !{metadata !"Color", i32 1, <4 x float>* @Color_typeProxy, metadata !8}
!8 = metadata !{i32 0, i32 0, i32 1027, null, i32 1}


Bottom IR:
; ModuleID = 'Glslang'

@gl_FragDepth = global float 0.000000e+00
@Depth = global float 0.000000e+00
@gl_FragColor = global <4 x float> zeroinitializer
@Color = global <4 x float> zeroinitializer

define fastcc void @main() {
entry:
  %gl_FragDepth = load float* @Depth, align 4
  store float %gl_FragDepth, float* @gl_FragDepth, align 4
  %gl_FragColor = load <4 x float>* @Color, align 16
  store <4 x float> %gl_FragColor, <4 x float>* @gl_FragColor, align 16
  br label %stage-epilogue

stage-epilogue:                                   ; preds = %entry
  br label %stage-exit

stage-exit:                                       ; preds = %stage-epilogue
  ret void
}

!gla.entrypoint = !{!0}
!gla.outputs = !{!1, !3}
!gla.inputs = !{!5, !7}

!0 = metadata !{metadata !"main", i32 15}
!1 = metadata !{metadata !"gl_FragDepth", i32 11, float* @gl_FragDepth_typeProxy, metadata !2}
!2 = metadata !{i32 0, i32 0, i32 1024, null, i32 0}
!3 = metadata !{metadata !"gl_FragColor", i32 7, <4 x float>* @gl_FragColor_typeProxy, metadata !4}
!4 = metadata !{i32 0, i32 0, i32 1026, null, i32 0}
!5 = metadata !{metadata !"Depth", i32 1, float* @Depth_typeProxy, metadata !6}
!6 = metadata !{i32 0, i32 0, i32 1025, null, i32 1}
!7 = metadata !{metadata !"Color", i32 1, <4 x float>* @Color_typeProxy, metadata !8}
!8 = metadata !{i32 0, i32 0, i32 1027, null, i32 1}
#version 130
// LunarGOO output
in float Depth;
in vec4 Color;

void main()
{
    gl_FragDepth = Depth;
    gl_FragColor = Color;
    
}


#version 130
// LunarGOO output
in float Depth;
in vec4 Color;

void main()
{
    gl_FragDepth = Depth;
    gl_FragColor = Color;
    
}


